### PR TITLE
[T7][F09-C4]

### PR DESCRIPTION
**Pull request by githubxxcc**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -70,16 +71,16 @@ Each of the four components
...
-<img src="images\SDforDeletePerson.png" width="800">
+<img src="images\SDforDeleteTask.png" width="800">
 
 >Note how the `Model` simply raises a `ModelChangedEvent` when the model is changed,
  instead of asking the `Storage` to save the updates to the hard disk.
 
 The diagram below shows how the `EventsCenter` reacts to that event, which eventually results in the updates
 being saved to the hard disk and the status bar of the UI being updated to reflect the 'Last Updated' time. <br>
-<img src="images\SDforDeletePersonEventHandling.png" width="800">
+<img src="images\SDforDeleteTaskEventHandling.png" width="800">
```
Comment (chanjunweimy) [inline docs/DeveloperGuide.md]: The sequence diagram here has a bit of problem. First of all there should be an activation bar for the "Post()" Event, 2ndly, if your second diagram has double flow then it's not so good, you need to think of a way to improve it. :)

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -243,54 +244,244 @@ a. Include those libraries in the repo (this bloats the repo size)<br>
...
+Extensions
+1a. There is no command which modified the TaskBook during this session
+  > 1a1. Taskbook displays displays a message indicating no commands can be undone
+  Use case ends
+
 {More to be added}
 
 ## Appendix C : Non Functional Requirements
+- Storage
+  - Should not use relational databases. Data storage must be done using text files you create yourself.
```
Comment (chanjunweimy) [inline docs/DeveloperGuide.md]: not only text files.. It could be json, xml....

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -243,54 +244,244 @@ a. Include those libraries in the repo (this bloats the repo size)<br>
...
+- Should work on a desktop without network/Internet connection. It should not be a mobile app or a cloud-based application. 
+- Should work stand-alone. It should not be a plug-in to another software. 
+- Should follow the Object-oriented paradigm. 
+- Should work without requiring an installer. Having an optional installer is OK as longs as the portable (non-installed) version has all the critical functionality. 
+- Should only work with Third Party User/Libraries if they,
+are 
+  - free.
+  - do not require any installation by the user of your software.
+  - do not violate other constraints.
+- Should not work with third-party software that can interfere with the learning objectives of the module.
```
Comment (chanjunweimy) [inline docs/DeveloperGuide.md]: This should be the requirement of the module, not non-functional requirement.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (githubxxcc) [inline docs/DeveloperGuide.md]: will work on it in later versions. Thanks

**Pull request closed without merging**